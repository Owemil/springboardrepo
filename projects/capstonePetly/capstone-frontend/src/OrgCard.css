.nearby-link {
    width: 18rem;
    height: 18rem;
    margin-top: 1rem;
}