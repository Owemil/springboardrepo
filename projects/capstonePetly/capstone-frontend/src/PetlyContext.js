import React from "react";

const PetlyContext = React.createContext();

export default PetlyContext;