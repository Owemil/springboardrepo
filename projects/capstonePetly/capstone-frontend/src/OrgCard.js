import React, { useContext } from "react";
import { Card, CardBody, CardTitle, List, ListGroupItem, NavLink, Button, Nav, Spinner } from "reactstrap";
import PetlyContext from "./PetlyContext";
import Favorite from "./Favorite";
import "./OrgCard.css"

// OrgCard component takes in org data for a specific org and data from any org that is favorited

//it renders a card with some info about the org

function OrgCard({ org, fav }) {
    const { user, getOrgAnimals } = useContext(PetlyContext)
    let orgData
    if (org) {
        orgData = {
            orgId: org.id,
            orgName: org.attributes.name,
            orgUrl: org.attributes.url,
            orgContact: org.attributes.email ? org.attributes.email : org.attributes.phone
        }
    }

    let favOrgs = user ? user.favOrgs.map(orgs => orgs.orgId) : []


    const handleClick = async () => {
        if (org) {
            return await getOrgAnimals(org.id, org.attributes.name)
        } else if (fav) {
            return await getOrgAnimals(fav.orgId, fav.orgName)
        }
    }

    return (
        <>

            {fav ? <Card
                className="nearby-link"

            >

                <Favorite
                    org={fav}
                    currFav={true} />


                <CardBody>
                    <CardTitle>
                        <Nav>
                            <NavLink href={fav.orgUrl}>
                                {fav.orgName}
                            </NavLink>

                        </Nav>
                    </CardTitle>
                    <List>
                        <ListGroupItem>
                            {fav.orgName}
                        </ListGroupItem>
                        <ListGroupItem>
                            {fav.orgUrl}
                        </ListGroupItem>
                        <ListGroupItem>
                            {fav.orgContact}
                        </ListGroupItem>
                    </List>
                    <Button onClick={handleClick} color="primary">See Our Pets!</Button>
                </CardBody>
            </Card>
                : org ?
                    <Card
                        className="nearby-link"

                    >
                        {favOrgs.includes(+org.id) ?
                            <Favorite
                                org={orgData}
                                currFav={true} />
                            :
                            <Favorite
                                org={orgData}
                            />}

                        <CardBody>
                            <CardTitle>
                                <Nav>
                                    <NavLink href={org.attributes.url}>
                                        {org.attributes.name}
                                    </NavLink>

                                </Nav>
                            </CardTitle>
                            <List>
                                <ListGroupItem>
                                    located In: {org.attributes.citystate}
                                </ListGroupItem>
                                <ListGroupItem>
                                    Email: {org.attributes.email}
                                </ListGroupItem>
                                <ListGroupItem>
                                    {org.attributes.distance} Miles Away
                                </ListGroupItem>
                            </List>
                            <Button onClick={handleClick} color="primary">See Our Pets!</Button>
                        </CardBody>
                    </Card>
                    : <div> <p>Loading...</p><Spinner color="info" /></div>}
        </>

    )

}

export default OrgCard