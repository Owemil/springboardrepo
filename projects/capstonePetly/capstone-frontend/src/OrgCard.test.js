import React, { useContext } from "react";
import OrgCard from "./OrgCard.js"
import PetlyContext from "./PetlyContext.js";
import { fireEvent, render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

const user = { favOrgs: [{ orgId: 3 }, { orgId: 2 }] }

const getOrgAnimals = (id, name) => {

}

const org = {
    id: 1,
    attributes: {
        name: "Tacoma Humane Society",
        url: "www.rescue.com",
        email: "contact@email.com",
        citystate: "Tacoma",
        distance: "2.36"

    }
}

const fav = {
    orgId: 2,
    orgName: "Seattle Rescue",
    orgUrl: "www.rescue.com",
    orgContact: "123-456-7890"
}

it("should render without crashing", function () {
    render(
        <MemoryRouter>
            <PetlyContext.Provider value={{
                user: user,
                getOrgAnimals: getOrgAnimals

            }}>
                <OrgCard />
            </PetlyContext.Provider>
        </MemoryRouter>)
})

it("matches snapshots", function () {

    const { asFragment } = render(
        <MemoryRouter>
            <PetlyContext.Provider value={{
                user: user,
                getOrgAnimals: getOrgAnimals

            }}>
                <OrgCard />
            </PetlyContext.Provider>
        </MemoryRouter>
    )

    expect(asFragment).toMatchSnapshot()
})
it("matches snapshots", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <PetlyContext.Provider value={{
                user: user,
                getOrgAnimals: getOrgAnimals

            }}>
                <OrgCard org={org} />
            </PetlyContext.Provider>
        </MemoryRouter>)

    expect(asFragment).toMatchSnapshot()
})
it("matches snapshots", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <PetlyContext.Provider value={{
                user: user,
                getOrgAnimals: getOrgAnimals

            }}>
                <OrgCard fav={fav} />
            </PetlyContext.Provider>
        </MemoryRouter>)

    expect(asFragment).toMatchSnapshot()
})

// it("reroutes (org prop)", function () {
//     const { getByText } = render(
//         <MemoryRouter>
//             <PetlyContext.Provider value={{
//                 user: user,
//                 getOrgAnimals: getOrgAnimals

//             }}>
//                 <OrgCard org={org} />
//             </PetlyContext.Provider>
//         </MemoryRouter>)

//     const btn = getByText("See Our Pets!")

//     fireEvent.click(btn)

//     expect(getByText("Rerouted to /Tacoma Humane Society/1/animals")).toBeInTheDocument()
// })

// it("reroutes (fav prop)", function () {
//     const { getByText } = render(
//         <MemoryRouter>
//             <PetlyContext.Provider value={{
//                 user: user,
//                 getOrgAnimals: getOrgAnimals

//             }}>
//                 <OrgCard fav={fav} />
//             </PetlyContext.Provider>
//         </MemoryRouter>)

//     const btn = getByText("See Our Pets!")

//     fireEvent.click(btn)


//     expect(getByText("Rerouted to /Seattle Rescue/2/animals")).toBeInTheDocument()


// })