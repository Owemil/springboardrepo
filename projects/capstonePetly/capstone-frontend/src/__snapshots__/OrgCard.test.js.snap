// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots 1`] = `[Function]`;

exports[`matches snapshots 2`] = `[Function]`;

exports[`matches snapshots 3`] = `[Function]`;
